.github/workflows/main.yml

```diff
@@ -21,6 +23,9 @@ jobs:
           check-latest: true
           cache: npm
 
+      - name: Set version in package.json using `git describe`
+        run: npm version --no-git-tag-version $(git describe --always)
```

Comment:
Running `git describe --always` locally results in a string like `v2.0.0-alpha.1-95-g517d5cc0`, which is a great unique identifier, but it is not a valid [semantic version](https://semver.org/). It does seem to work, though, and `npm version` trims the invalid parts of the string, so I guess it is fine?

That said, we should try and skip this step when we're doing a tagged release, can we make this conditional like the `publish-npm` job?

Reply:
`git describe` on tag v1.9.4 yields `v1.9.4` (as intended), so we should be fine?

```
> git checkout v1.9.4
HEAD is now at d15112c9 1.9.4

> git describe --always
v1.9.4
```